Content.Server/Implants/RadioImplantSystem.cs

```diff
@@ -0,0 +1,77 @@
+﻿using Robust.Shared.Containers;
+using Content.Shared.FixedPoint;
+using Content.Shared.Implants;
+using Content.Shared.Implants.Components;
+using Content.Shared.Tag;
+using Content.Server.Radio.Components;
+
+namespace Content.Server.Implants;
+
+public sealed class RadioImplantSystem : EntitySystem
+{
+    public override void Initialize()
+    {
+        base.Initialize();
+        SubscribeLocalEvent<RadioImplantComponent, ImplantImplantedEvent>(OnImplantImplantedEvent);
+        SubscribeLocalEvent<RadioImplantComponent, EntGotRemovedFromContainerMessage>(OnRemove);
+    }
+
+    /// <summary>
+    /// If implanted with a radio implant, installs the necessary intrinsic radio components
+    /// </summary>
+    private void OnImplantImplantedEvent(Entity<RadioImplantComponent> ent, ref ImplantImplantedEvent args)
+    {
+        if (args.Implanted == null)
+            return;
+
+        var activeRadio = EnsureComp<ActiveRadioComponent>(args.Implanted.Value);
+        foreach (var channel in ent.Comp.RadioChannels)
+        {
+            if (activeRadio.Channels.Add(channel))
+                ent.Comp.ActiveAddedChannels.Add(channel);
+        }
+
+        EnsureComp<IntrinsicRadioReceiverComponent>(args.Implanted.Value);
+
+        var intrinsicRadioTransmitter = EnsureComp<IntrinsicRadioTransmitterComponent>(args.Implanted.Value);
+        foreach (var channel in ent.Comp.RadioChannels)
+        {
+            if (intrinsicRadioTransmitter.Channels.Add(channel))
+                ent.Comp.TransmitterAddedChannels.Add(channel);
+        }
+    }
+
+    /// <summary>
+    /// Removes intrinsic radio components once the Radio Implant is removed
+    /// </summary>
+    private void OnRemove(Entity<RadioImplantComponent> ent, ref EntGotRemovedFromContainerMessage args)
+    {
+        if (TryComp<ActiveRadioComponent>(args.Container.Owner, out var activeRadioComponent))
+        {
+            foreach (var channel in ent.Comp.ActiveAddedChannels)
+            {
+                activeRadioComponent.Channels.Remove(channel);
+            }
+            ent.Comp.ActiveAddedChannels.Clear();
+
+            if (activeRadioComponent.Channels.Count == FixedPoint2.Zero)
+            {
+                RemCompDeferred<ActiveRadioComponent>(args.Container.Owner);
+            }
+        }
+
+        if (!TryComp<IntrinsicRadioTransmitterComponent>(args.Container.Owner, out var radioTransmitterComponent))
+            return;
+
+        foreach (var channel in ent.Comp.TransmitterAddedChannels)
+        {
+            radioTransmitterComponent.Channels.Remove(channel);
+        }
+        ent.Comp.TransmitterAddedChannels.Clear();
+
+        if (radioTransmitterComponent is { Channels.Count: 0 } || activeRadioComponent is { Channels.Count: 0 })
```

Comment:
```suggestion
        if (radioTransmitterComponent.Channels.Count == 0 || activeRadioComponent.Channels.Count == 0 )
```
Let's keep it consistent with the way done above.